presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -506,6 +506,10 @@ private static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inpu
             return Optional.empty();
         }
 
+        if (blockLocations.length == 0) {
+            return Optional.empty();
+        }
```

Comment:
@haozhun i was unsure if this condition should be placed here or below the loop which filters `BlockLocation` based on `start` and `length`. For the sake of empty files, this is no difference, but -- maybe -- we can arrive at empty `List<InternalHiveBlock> blocks` even when block locations isn't empty?

Reply:
It would be nice to add some unit tests. Relying on product tests causes a long development cycle.

Reply:
Yup, but I couldn't reproduce the issue without using `hive` to load empty file onto HDFS. @kokosing  how can I do this?

Reply:
I can reproduce the problem with such a unit test (I think you can add this to`TestBackgroundSplitLoader`) and your fix fixes it:

```
    @Test
    public void testEmptySplit()
            throws Exception
    {
        BackgroundHiveSplitLoader backgroundHiveSplitLoader = backgroundHiveSplitLoader(
                ImmutableList.of(emptySplit(RETURNED_PATH)),
                TupleDomain.none());

        HiveSplitSource hiveSplitSource = hiveSplitSource(backgroundHiveSplitLoader, TupleDomain.none());
        backgroundHiveSplitLoader.start(hiveSplitSource);

        assertEquals(drain(hiveSplitSource).size(), 0);
    }
```

where

```
    private static LocatedFileStatus emptySplit(Path path)
    {
        return new LocatedFileStatus(
                0L,
                false,
                0,
                0L,
                0L,
                0L,
                null,
                null,
                null,
                null,
                path,
                new BlockLocation[] {});
    }
```

Reply:
@nezihyigitbasi thanks!